src/vm/jitinterface.cpp

```diff
@@ -7251,6 +7251,42 @@ bool getILIntrinsicImplementationForUnsafe(MethodDesc * ftn,
         methInfo->options = (CorInfoOptions)0;
         return true;
     }
+    else if (tk == MscorlibBinder::GetMethod(METHOD__UNSAFE__BYREF_NULLREF)->GetMemberDef())
+    {
+        static const BYTE ilcode[] = { CEE_LDNULL, CEE_RET };
```

Comment:
This has the same type mismatch as what you have hinted on below. This should be `CEE_LDC_I4_0, CEE_CONV_U, CEE_RET`.